src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogApiClient.java

```diff
@@ -314,18 +315,91 @@ private void postApi(final JSONObject payload, final String type) throws IOExcep
+        private final HttpClient httpClient;
+
+        private final CircuitBreaker<List<String>> circuitBreaker;
+
+        public ApiLogWriteStrategy(String logIntakeUrl, Secret apiKey, HttpClient httpClient) {
+            this.logIntakeUrl = logIntakeUrl;
+            this.apiKey = apiKey;
+            this.httpClient = httpClient;
+            this.circuitBreaker = new CircuitBreaker<>(
+                    this::doSend,
+                    this::fallback,
+                    this::handleError,
+                    100,
+                    CircuitBreaker.DEFAULT_MAX_HEALTH_CHECK_DELAY_MILLIS,
+                    CircuitBreaker.DEFAULT_DELAY_FACTOR);
+        }
+
+        @Override
+        public void send(List<String> logs) {
+            circuitBreaker.accept(logs);
+        }
+
+        private void doSend(List<String> payloads) throws Exception {
+            Map<String, String> headers = new HashMap<>();
+            headers.put("DD-API-KEY", Secret.toString(apiKey));
+            headers.put("Content-Encoding", "gzip");
+
+            ByteArrayOutputStream request = new ByteArrayOutputStream();
+            GZIPOutputStream compressedRequest = new GZIPOutputStream(request);
+            int uncompressedRequestLength = 0;
+
+            for (String payload : payloads) {
+                byte[] body = payload.getBytes(StandardCharsets.UTF_8);
+                if (body.length + 2 > PAYLOAD_SIZE_LIMIT) { // + 2 is for array beginning and end: [<payload>]
+                    logger.severe("Dropping a log because payload size (" + body.length + ") exceeds the allowed limit of " + PAYLOAD_SIZE_LIMIT);
+                    continue;
+                }
+
+                if (uncompressedRequestLength + body.length + 2 > PAYLOAD_SIZE_LIMIT) { // + 2 is for comma and array end: ,<payload>]
+                    compressedRequest.write(END_JSON_ARRAY);
+                    compressedRequest.close();
+                    httpClient.post(logIntakeUrl, headers, "application/json", request.toByteArray(), Function.identity());
+                    request = new ByteArrayOutputStream();
+                    compressedRequest = new GZIPOutputStream(request);
+                    uncompressedRequestLength = 0;
+                }
+
+                compressedRequest.write(uncompressedRequestLength == 0 ? BEGIN_JSON_ARRAY : COMMA);
+                compressedRequest.write(body);
+                uncompressedRequestLength += body.length + 1;
+            }
+
+            compressedRequest.write(END_JSON_ARRAY);
+            compressedRequest.close();
+            httpClient.post(logIntakeUrl, headers, "application/json", request.toByteArray(), Function.identity());
+        }
+
+        private void handleError(Exception e) {
+            DatadogUtilities.severe(logger, e, "Failed to post logs");
+        }
+
+        private void fallback(List<String> payloads) {
+            // cannot establish connection to agent, do nothing
```

Comment:
Thanks, the comment was a copy-paste that I overlooked.
The first error will logged in `org.datadog.jenkins.plugins.datadog.clients.DatadogApiClient.ApiLogWriteStrategy#handleError` above, which is invoked when the circuit breaker goes from active to inactive state.
In the fallback method we shouldn't be logging it, otherwise we'll spam the logs